pkg/controllers/maintenancelimit_controller.go

```diff
@@ -111,3 +94,89 @@ func (r *MaintenanceLimitReconciler) UpdateMaintenanceLimits(ctx context.Context
 
 	return nil
 }
+
+// getMaintenanceLimitsByType calculates the max number of nodes that can be updated at the given time
+func (r *MaintenanceLimitReconciler) getMaintenanceLimitsByType(ctx context.Context, log logr.Logger, ml *repairmanv1.MaintenanceLimit) (uint, error) {
+	if !strings.EqualFold(ml.Name, "node") {
+		return 0, errors.New("unsupported maintenance type")
+	}
+
+	var policied map[string]bool
+	err := r.getPoliciedResourcesByType(ctx, log, ml, policied)
+	if err != nil {
+		log.Error(err, "unable to get policied resource", "type", ml.Name)
+		return 0, err
+	}
+
+	var availableResources uint
+	switch ml.Name {
+	case "node":
```

Comment:
Trying to keep my code extensible/flexible for when we add other resources. Was this a good way to do it? Is there a better way?